dcore/diagnostics.py

```diff
@@ -0,0 +1,19 @@
+from firedrake import assemble, dot, dx
+from math import sqrt
+
+
+class Diagnostics(object):
+
+    def __init__(self, fieldlist=None):
```

Comment:
You might want to check that the to be registered fields are not already in the existing fields.

```python
def register(self, *fields):
    fset = set(self.fields)
    for f in fields:
        if f not in fset:
            self.fields.append(f)
```